client/src/components/books/books/BooksModal.1.js

```diff
@@ -0,0 +1,349 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Document, Page } from 'react-pdf';
+import cloudinary from 'cloudinary';
+import { saveBooks, addBook } from '../../../actions/books';
+import Modal from '../../modal/Modal';
+import Button from '../../form/Button';
+import InputField from '../../form/InputField';
+import Textarea from '../../form/TextareaField';
+import FlashMessagesList from '../../flash/FlashMessagesList';
+import { addFlashMessage } from '../../../actions/flashMessages';
+import UploadBookCover from './UploadBookCover';
+
+const $ = window.$;
+
+/* eslint-disable require-jsdoc, class-methods-use-this */
+class BooksModal extends Component {
+    state = {
```

Comment:
The structure of your state can be better. You might consider extracting book properties into a key so that it's easier to update the state.

Reply:
Hello Femi, 

Thank you for the feedback.

I once tried putting it in a book object in the state, like you mentioned but the challenge was that I couldn't use a single onChange event to control the form fields.

I will look into it again and give it another shot.

Thanks once again

Reply:
Hi Femi,

I have figured it out.

Thank you for the feedback